staging/kos/pkg/controllers/connectionagent/connection_agent.go

```diff
@@ -1052,15 +1054,26 @@ func (ca *ConnectionAgent) localAttachmentIsUpToDate(att *netv1a1.NetworkAttachm
 }
 
 func (ca *ConnectionAgent) updateLocalAttachmentStatus(att *netv1a1.NetworkAttachment, macAddr, ifcName string, statusErrs sliceOfString, pcer *netv1a1.ExecReport) error {
-	att2 := att.DeepCopy()
-	att2.Status.MACAddress = macAddr
-	att2.Status.IfcName = ifcName
-	att2.Status.HostIP = ca.hostIP.String()
-	att2.Status.Errors.Host = statusErrs
-	att2.Status.PostCreateExecReport = pcer
+	status := netv1a1.NetworkAttachmentStatus{
+		MACAddress:           macAddr,
+		IfcName:              ifcName,
+		HostIP:               ca.hostIP.String(),
+		Errors:               netv1a1.NetworkAttachmentErrors{Host: statusErrs},
+		PostCreateExecReport: pcer,
+	}
+	var pb bytes.Buffer
+	pb.WriteString(`{"status":`)
+	encoder := json.NewEncoder(&pb)
```

Comment:
Just asking: isn't there something at a higher-level? API objects are marshalled to Go structs to free controllers from playing with JSON strings; re-introducing code that directly deals with JSON strings is inelegant.

Reply:
I am not aware of anything higher level for patches --- which are inherently about fragments of objects.  The higher level stuff that I know about is for whole objects.